File: packages/ts/test/__fixtures__/interpret/dirsql.config_raises.mjs
// Fixture whose `extract` raises -- exercises the `ok: false` response.

import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { DirSQL } from "dirsql";

const here = dirname(fileURLToPath(import.meta.url));

export default new DirSQL({
  root: join(here, "data"),
  tables: [
    {
      ddl: "CREATE TABLE papers (title TEXT)",
      glob: "**/meta.json",
      extract: () => {
        throw new Error("synthetic extract failure");
      },
    },
  ],
});

File: packages/python/tests/e2e/__fixtures__/interpret/dirsql.config_raises.py
"""Fixture whose `extract` raises -- exercises the `ok: false` response."""

import os

from dirsql import DirSQL, Table


def _boom(_path):
    raise ValueError("synthetic extract failure")


app = DirSQL(
    root=os.path.join(os.path.dirname(__file__), "data"),
    tables=[
        Table(
            ddl="CREATE TABLE papers (title TEXT)",
            glob="**/meta.json",
            extract=_boom,
        )
    ],
)
